vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -510,12 +510,14 @@ def collect_events(self):
     def check(self, _):
         # type: (Any) -> None
         self._hostname = datadog_agent.get_hostname()
-        # Assert the health of the vCenter API and submit the service_check accordingly
+        # Assert the health of the vCenter API by getting the version, and submit the service_check accordingly
         try:
-            self.api.check_health()
+            version_info = self.api.get_version()
+            if self.is_metadata_collection_enabled():
```

Comment:
given that this also checks if the provided endpoint is of the right type maybe it should always happen (the config option will prevent the agent from submitting the version value).

Reply:
The part where it checks if the endpoint is of the right type always happen (upon connection), I'm just limiting the submission of metadata to the agent here.